kymatio/scattering1d/core/timefrequency_scattering.py

```diff
@@ -103,14 +103,18 @@ def timefrequency_scattering(
             k2 = max(j2 - k1 - oversampling, 0)  # what we subsample now in 2nd
             Y_2_c = B.cdgmm(U_1_hat, psi2[n2][k1])
             Y_2_hat = B.subsample_fourier(Y_2_c, 2**k2)
-            Y_2_list.append(B.ifft(Y_2_hat))
+            Y_2_c = B.ifft(Y_2_hat)
+
+            if Y_2_arr is None:
+                Y_2_arr = backend.zeros((2**sc_freq.J_pad[n2], Y_2_c.shape[-1]),
+                                        dtype=Y_2_c.dtype)
+            Y_2_arr[n1 + sc_freq.pad_left[n2]] = Y_2_c
 
         # sum is same for all `n1`, just take last
         k1_plus_k2 = k1 + k2
 
-        # zero-pad along frequency, map to Fourier domain
-        total_height = 2 ** sc_freq.J_pad
-        Y_2_hat = _pad_transpose_fft(Y_2_list, total_height, B, B.fft)
+        # zero-pad along frequency, map to Fourier domain  # TODO comment
```

Comment:
TODO comment ?

Reply:
More like `# TODO improve comment`, i.e. make clearer

Reply:
that already seems quite clear to me honestly :)